appinventor/components/src/com/google/appinventor/components/runtime/ButtonBase.java

```diff
@@ -169,16 +169,22 @@ public boolean onTouch(View view, MotionEvent me) {
       if (me.getAction() == MotionEvent.ACTION_DOWN) {
         //button pressed, provide visual feedback AND return false
         if (ShowFeedback() && (AppInventorCompatActivity.isClassicMode() || Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP)) {
-          view.getBackground().setAlpha(70); // translucent
-          view.invalidate();
+          if (backgroundImageDrawable != null || imagePath != "") {
+            view.getBackground().setAlpha(70); // translucent
```

Comment:
I think that you should still be invalidating the view because of the change in the alpha here. Compared with the old version, the view is only invalidated if it's not changed, which doesn't seem right to me.

Reply:
Yes but that wouldn't really work well against Material Design, it's already going against it by just setting a different alpha if there's an image. Unfortunately at this point, I don't think there is any way to warn users of a possible removal of the image property on the button. The Image Component is meant for images. This would make it easier to remove the code which I've added to check the image state, and instead just keep invalidating it.